google-cloud-spanner-cassandra/src/main/java/com/google/cloud/spanner/adapter/PreparePayloadResult.java

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2025 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.google.cloud.spanner.adapter;
+
+import com.google.api.gax.rpc.ApiCallContext;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An object used to encapsulate the result of preparing the Adapter payload prior to sending the
+ * gRPC request.
+ */
+public class PreparePayloadResult {
+  private Optional<byte[]> attachmentErrorResponse;
+  private ApiCallContext context;
+  private Map<String, String> attachments;
+
+  public PreparePayloadResult(
+      Map<String, String> attachments,
+      Optional<byte[]> attachmentErrorResponse,
```

Comment:
instead of Optional, does it make sense to have another variant of ctor with just (context, attachments)?

Reply:
Something like: 
```
public PreparePayloadResult(context, attachments, attachmentErrorResponse) { }

public PreparePayloadResult(context, attachments) { this(context, attachments,  Optional.empty()}

public PreparePayloadResult(context) { this(context, new HashMap(),  Optional.empty()}

 ```

Reply:
I realize the attachments map may be created regardless of the type (for example when setting the max_commit_delay flag in a future PR) so it will be passed into the helper methods and only two constructors are needed.